### PR TITLE
Skip test requiring rpmdb

### PR DESCRIPTION
**Pull request by carlsmedstad**
Hey again! :wave:

A test was added without the proper predicate in the 2.1.0 release which causes it to fail on Arch Linux.

The added CI job should make sure this does not happen anymore in the future.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -7,7 +7,45 @@ on:
...
+              python3-pytest-xdist
+              python3-flake8
+              python3-flake8-import-order
+              python3-pyxdg
+              python3-toml
+              python3-zstd
+              python3-pip
+              rpm-build
+              git
+      - run: rm -rf $(rpm --eval %_dbpath)
```
Comment (Conan-Kudo) [inline .github/workflows/main.yml:44]: Please quote `%_dbpath` here so that the shell doesn't evaluate it.

**Review by Conan-Kudo — CHANGES_REQUESTED** [review]

**Review by marxin — CHANGES_REQUESTED** [review]
As mentioned, I prefer using matrix with `container` and `build-type` properties, where the later can be one of `normal`, `optional-deps`.

File: .github/workflows/main.yml (hunk 1 above)
Comment (carlsmedstad) [reply .github/workflows/main.yml]: Fixed!

**Comment by carlsmedstad** [comment]
> As mentioned, I prefer using matrix with `container` and `build-type` properties, where the later can be one of `normal`, `optional-deps`.

Something like in https://github.com/rpm-software-management/rpmlint/pull/690/commits/3cfe85ba58f2cd077c2a7111149b2f2df9326845?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -17,6 +17,9 @@ jobs:
           - 'registry.fedoraproject.org/fedora:rawhide'
           - 'registry.opensuse.org/opensuse/leap-dnf:15.3'
           - 'registry.opensuse.org/opensuse/tumbleweed-dnf:latest'
+        build-type:
+          - 'normal'
+          - 'no-optional-deps'
```
Comment (marxin) [inline .github/workflows/main.yml:22]: Nice! I would not run all 2x4 combinations, but I would rather use `include` and run only 2 `no-optional-deps` (one fedora and one opensuse).

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -52,26 +52,26 @@ jobs:
               rpm-build
               git
         if: ${{ contains(matrix.docker, 'opensuse') }}
+      - run: dnf --assumeyes install
+              checkbashisms dash
+              desktop-file-utils
+              appstream-glib
+              myspell-en_US myspell-cs_CZ
+              python3-pyenchant
+        if: ${{ contains(matrix.docker, 'opensuse') && contains(matrix.build-type, 'normal') }}
```
Comment (marxin) [inline .github/workflows/main.yml:61]: Use rather `matrix.build-type == 'normal'`, please.

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -88,6 +88,17 @@ jobs:
...
         if: ${{ contains(matrix.docker, 'fedora') }}
+      - run: dnf --nogpgcheck --assumeyes install
+              /usr/bin/appstream-util
+              /usr/bin/desktop-file-validate
+              dash
+              devscripts-checkbashisms
+              hunspell-en
+              hunspell-cs
+              python3-enchant
+        if: ${{ contains(matrix.docker, 'fedora') && contains(matrix.build-type, 'normal') }}
```
Comment (marxin) [inline .github/workflows/main.yml:99]: Likewise here.

**Review by marxin — CHANGES_REQUESTED** [review]
Very nice improvement, please fix the remaining comments I have.

**Comment by carlsmedstad** [comment]
I believe I've addressed all the comments. Also, I've squashed any redundant commits.

OpenSUSE's docker registry seems to be down thought, so those tests are failing.

**Comment by marxin** [comment]
> I believe I've addressed all the comments. Also, I've squashed any redundant commits.

Great, thank you for the contribution.

**Pull request merged**